pyiron_base/project/generic.py

```diff
@@ -1036,6 +1065,41 @@ def refresh_job_status_based_on_job_id(self, job_id, que_mode=True):
                 if not self.queue_check_job_is_waiting_or_running(job):
                     self.db.set_job_status(job_id=job_id, status="aborted")
 
+    def _refresh_job_status_file_table(self, df):
+        """
+        Internal function to refresh the job table and update the job table with the status from the queuing system.
+
+        Args:
+            df (pandas.DataFrame): job table from the file based database
+
+        Returns:
+            pandas.DataFrame: updated job table with status from the queuing system
+        """
+
+        def convert_queue_status(queue_status):
+            convert_status_dict = {"pending": "submitted"}
+            if queue_status in convert_status_dict.keys():
+                return convert_status_dict[queue_status]
+            else:
+                return queue_status
+
+        df_queue = state.queue_adapter.get_status_of_my_jobs()
+
+        status_lst = df.status.values.tolist()
+        working_dir_lst = df.project + df.job + "_hdf5/" + df.job
+        for i, [working_dir, status] in enumerate(
+            zip(working_dir_lst, status_lst.copy())
+        ):
+            if status == "initialized":
+                df_tmp = df_queue[df_queue.working_directory == working_dir]
+                if len(df_tmp) > 0:
+                    status_lst[i] = convert_queue_status(
+                        queue_status=df_tmp.status.values[0]
+                    )
+        df["status"] = status_lst
+
+        return df
+
```

Comment:
If I understand this correctly, this modifies a dataframe obtained from `FileTable.job_table` and sets the updated job status there and returns it.  But this won't modify the dataframe `FileTable` keeps internally, so the queuing system would be queried every time `Project.job_table` is called, right?  Couldn't you equivalently call `FileTable.set_job_status` in line 1096?

Reply:
`set_job_status` writes to the HDF5 files which means we would open the HDF5 files in write access on a shared file system, which can cause errors in high throughput settings. If the job fails and we catch the error then the pyiron process which is executing the job should automatically update the status, that is what I changed in https://github.com/pyiron/pyiron_base/pull/973 so for performance reasons I do not want to write to the HDF5 files here.